<!DOCTYPE html>
<html lang="es">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Clase 35 - Async-Await</title>
</head>
<body>
    <script
    src="https://code.jquery.com/jquery-3.5.1.min.js"
    integrity="sha256-9/aliU8dGd2tb6OSsuzixeV4y/faTqgFtohetphbbj0="
    crossorigin="anonymous"></script>
    <script>
        const API_URL = 'https://swapi.dev/api/';
        const PEOPLE_URL = 'people/:id';
        const OPTS = { crossDomain: true };
        
        
        function obtenerPersonaje(id) {

            return new Promise((resolve, reject) => {
                const URL = `${API_URL}${PEOPLE_URL.replace(':id', id)}`;
                $
                .get(URL, OPTS, function (data) {
                    resolve(data);
                })
                .fail( () => reject(id));
            });
        }

        function onError(id) {
            console.log(`SucediÃ³ un error al obtener el personaje ${id}.`);
        }

        async function obtenerPersonajes() {
            var ids = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
            var promesas = ids.map( id => obtenerPersonaje(id) );

            try {
                var personajes = await Promise.all(promesas);
                console.log(personajes);
            } catch (id) {
                onError(id);
            }
        }

        obtenerPersonajes();

    </script>
</body>
</html>